packages/install-expo-modules/src/plugins/ios/withIosModulesAppDelegate.ts

```diff
@@ -134,16 +134,91 @@ export function updateModulesAppDelegateSwift(
   contents: string,
   sdkVersion: string | undefined
 ): string {
-  if (sdkVersion && semver.lt(sdkVersion, '52.0.0')) {
-    return updateModulesAppDelegateSwiftLegacy(contents, sdkVersion);
+  if (sdkVersion) {
+    if (semver.lt(sdkVersion, '52.0.0')) {
+      return updateModulesAppDelegateSwiftLegacy(contents, sdkVersion);
+    }
+    if (semver.lt(sdkVersion, '53.0.0')) {
+      return updateModulesAppDelegateSwiftSdk52(contents, sdkVersion);
+    }
+  }
+
+  // Add imports if needed
+  if (!contents.match(/^import\s+Expo\s*$/m)) {
+    contents = addSwiftImports(contents, ['Expo']);
+  }
+
+  // Replace superclass with ExpoAppDelegate
+  contents = contents.replace(
+    /^(class\s+AppDelegate\s*:\s*)UIResponder,\s*UIApplicationDelegate(\W+)/m,
+    '$1ExpoAppDelegate$2'
+  );
+  // Remove non-overridable properties
+  contents = contents.replace(/^\s*var window: UIWindow\?\n/m, '');
+  contents = contents.replace(/^\s*var reactNativeDelegate: ReactNativeDelegate\?\n/m, '');
+  contents = contents.replace(/^\s*var reactNativeFactory: RCTReactNativeFactory\?\n/m, '');
+  // Add `override` keyword and return super call to didFinishLaunchingWithOptions
+  contents = contents.replace(
+    /\b(func application\([\s\S]+?didFinishLaunchingWithOptions launchOptions[\s\S]+?\{[\s\S]+?)(return true)([\s\S]+?\})/m,
+    'override $1return super.application(application, didFinishLaunchingWithOptions: launchOptions)$3'
+  );
+  // Remove implementation in didFinishLaunchingWithOptions,
+  // most of the code is not overridable from ExpoAppDelegate
+  contents = contents.replace(/^\s*let delegate = ReactNativeDelegate\(\)\n/m, '');
+  contents = contents.replace(
+    /^\s*let factory = RCTReactNativeFactory\(delegate: delegate\)\n/m,
+    ''
+  );
+  contents = contents.replace(/^\s*window = UIWindow\(frame: UIScreen\.main\.bounds\)\n/m, '');
+  contents = contents.replace(
+    /^\s*delegate.dependencyProvider = RCTAppDependencyProvider\(\)\n/m,
+    ''
+  );
+  contents = contents.replace(/^\s*reactNativeDelegate = delegate\n/m, '');
+  contents = contents.replace(/^\s*reactNativeFactory = factory\n/m, '');
+  contents = contents.replace(/^\s*window = UIWindow(frame: UIScreen.main.bounds)\n/m, '');
+  contents = contents.replace(
+    /^\s*factory\.startReactNative\([\s\S]+?withModuleName:\s*"(.+)",[\s\S]+?\)\n/m,
+    `\
+    self.moduleName = "$1"
+    self.initialProps = [:]`
+  );
```

Comment:
we basically replace the whole `didFinishLaunchingWithOptions` block. i don't want to replace the whole block at once, because some people may have their custom code inside `didFinishLaunchingWithOptions`. that results in this ugly replacement one by one.